vm/test.c

```diff
@@ -48,19 +49,27 @@ int main(int argc, char **argv)
         { .name = "help", .val = 'h', },
         { .name = "mem", .val = 'm', .has_arg=1 },
         { .name = "jit", .val = 'j' },
+        { .name = "inst", .val = 'i' },
         { .name = "register-offset", .val = 'r', .has_arg=1 },
         { }
     };
 
     const char *mem_filename = NULL;
     bool jit = false;
+    struct ubpf_inst_cnt *inst_cnt = NULL;
 
     int opt;
-    while ((opt = getopt_long(argc, argv, "hm:jr:", longopts, NULL)) != -1) {
+    while ((opt = getopt_long(argc, argv, "hm:ijr:", longopts, NULL)) != -1) {
         switch (opt) {
         case 'm':
             mem_filename = optarg;
             break;
+        case 'i':
+            inst_cnt = malloc(sizeof(struct ubpf_inst_cnt));
```

Comment:
isnt_cnt probably doesn't need to be allocated on the heap. It can easily be done on the stack. Saves the need to free it. So just set a bool here and conditionally call `ubpf_set_inst_cnt(vm, &inst_cnt);` later if the bool is set.